File: test/language/expressions/java/expressions/CastUInt64ToUInt8Test.java
package expressions;

import static org.junit.Assert.*;

import java.math.BigInteger;

import org.junit.Test;

import expressions.cast_uint64_to_uint8.CastUInt64ToUInt8Expression;

public class CastUInt64ToUInt8Test
{
    @Test
    public void uint8ValueUsingUInt64Value()
    {
        final BigInteger uint64Value = new BigInteger("FFFFFFFFFFFFFFFE", 16);
        final CastUInt64ToUInt8Expression castUInt64ToUInt8Expression =
                new CastUInt64ToUInt8Expression(uint64Value, false);
        final short expectedUInt8Value = uint64Value.shortValue();
        assertEquals(expectedUInt8Value, castUInt64ToUInt8Expression.uint8Value());
    }

    @Test
    public void uint8ValueUsingConstant()
    {
        final BigInteger uint64Value = new BigInteger("FFFFFFFFFFFFFFFE", 16);
        final CastUInt64ToUInt8Expression castUInt64ToUInt8Expression =
                new CastUInt64ToUInt8Expression(uint64Value, true);
        final short expectedUInt8Value = 1;
        assertEquals(expectedUInt8Value, castUInt64ToUInt8Expression.uint8Value());
    }
}

File: test/extensions/language/expressions/python/CastUInt64ToUInt8Test.py
import Expressions


class CastUInt64ToUInt8Test(Expressions.TestCase):
    def testUInt8ValueUsingUInt64Value(self):
        uint64Value = 0xFFFFFFFFFFFFFFFE
        castUInt64ToUInt8Expression = self.api.CastUInt64ToUInt8Expression(uint64Value, use_constant_=False)
        expectedUInt8Value = uint64Value
        self.assertEqual(expectedUInt8Value, castUInt64ToUInt8Expression.uint8_value())

    def testUint8ValueUsingConstant(self):
        uint64Value = 0xFFFFFFFFFFFFFFFE
        castUInt64ToUInt8Expression = self.api.CastUInt64ToUInt8Expression(uint64Value, True)
        expectedUInt8Value = 1
        self.assertEqual(expectedUInt8Value, castUInt64ToUInt8Expression.uint8_value())
